ddtrace/contrib/anthropic/patch.py

```diff
@@ -0,0 +1,205 @@
+    },
+)
+
+
+@with_traced_module
+def traced_chat_model_generate(anthropic, pin, func, instance, args, kwargs):
+    chat_messages = get_argument_value(args, kwargs, 0, "messages")
+    integration = anthropic._datadog_integration
+
+    operation_name = func.__name__
+
+    span = integration.trace(
+        pin,
+        "%s.%s" % (instance.__class__.__name__, operation_name),
+        submit_to_llmobs=True,
+        interface_type="chat_model",
+        provider="anthropic",
+        model=kwargs.get("model", ""),
+        api_key=_extract_api_key(instance),
+    )
+
+    chat_completions = None
+    try:
+        for message_idx, message in enumerate(chat_messages):
+            if not isinstance(message, dict):
+                continue
+            if isinstance(message.get("content", None), str):
+                if integration.is_pc_sampled_span(span):
+                    span.set_tag_str(
+                        "anthropic.request.messages.%d.content.0.text" % (message_idx),
+                        integration.trunc(message.get("content", "")),
+                    )
+                span.set_tag_str(
+                    "anthropic.request.messages.%d.content.0.type" % (message_idx),
+                    "text",
+                )
+            elif isinstance(message.get("content", None), list):
+                for block_idx, block in enumerate(message.get("content", [])):
+                    if integration.is_pc_sampled_span(span):
+                        if _get_attr(block, "type", None) == "text":
+                            span.set_tag_str(
+                                "anthropic.request.messages.%d.content.%d.text" % (message_idx, block_idx),
+                                integration.trunc(str(_get_attr(block, "text", ""))),
+                            )
+                        elif _get_attr(block, "type", None) == "image":
+                            span.set_tag_str(
+                                "anthropic.request.messages.%d.content.%d.text" % (message_idx, block_idx),
+                                "([IMAGE DETECTED])",
+                            )
+
+                    span.set_tag_str(
+                        "anthropic.request.messages.%d.content.%d.type" % (message_idx, block_idx),
+                        _get_attr(block, "type", "text"),
+                    )
+            span.set_tag_str(
+                "anthropic.request.messages.%d.role" % (message_idx),
+                message.get("role", ""),
+            )
+        tag_params_on_span(span, kwargs, integration)
+
+        chat_completions = func(*args, **kwargs)
+
+        if isinstance(chat_completions, anthropic.Stream) or isinstance(
+            chat_completions, anthropic.lib.streaming._messages.MessageStreamManager
+        ):
+            pass
+        else:
+            handle_non_streamed_response(integration, chat_completions, args, kwargs, span)
```

Comment:
Not a fan of this. Can we get rid of the `pass` and just make this an if statement without an else